spec/Observable-spec.js

```diff
@@ -166,6 +207,38 @@ describe('Observable', function () {
           Observable.empty().subscribe({});
         }).not.toThrow();
       });
+
+      it('should run unsubscription logic when an error is thrown sending messages synchronously to an anonymous observer', function () {
+        var messageError = false;
+        var messageErrorValue = false;
+        var unsubscribeCalled = false;
+
+        var o = {
+          next: function next(x) {
+            expect(this).toBe(o);
+            throw x;
+          }
+        };
+        var sub;
+        var source = new Observable(function (observer) {
+          observer.next('boo!');
+          return function () {
+            unsubscribeCalled = true;
+          };
+        });
+
+        try {
+          sub = source.subscribe(o);
+        } catch (e) {
+          messageError = true;
+          messageErrorValue = e;
+        }
+
+        expect(sub).toBe(undefined);
+        expect(unsubscribeCalled).toBe(true);
+        expect(messageError).toBe(true);
+        expect(messageErrorValue).toBe('boo!');
+      });
```

Comment:
Can we add an additional test to ensure that no more emissions occur once the next handler throws?

This shouldn't emit `3`, for example:

```js
let tornDown = false;

new Observable(observer => {
  observer.next(1);
  observer.next(2);
  observer.next(3);

  function () => {
    tornDown = true;
  }
})
.subscribe(x => {
  if (x === 2) {
    throw new Error('I hate twos');
  }
  console.log(x);
});
```